cps/kni.c

```diff
@@ -613,10 +661,138 @@ mnl_socket_recvfrom_flags(const struct mnl_socket *nl, void *buf, size_t bufsiz,
 	return ret;
 }
 
+static int
+new_route(struct route_update *update, struct cps_config *cps_conf)
+{
+	int ret;
+	uint16_t proto;
+	char ip_buf[INET6_ADDRSTRLEN];
+	char ipp_buf[INET6_ADDRSTRLEN + 4];
+	int gw_fib_id;
+	struct ip_prefix prefix_info;
+	struct ipaddr gw_addr;
+	struct gk_fib *gw_fib = NULL;
+	struct gk_lpm *ltbl = &cps_conf->gk->lpm_tbl;
+
+	if (update->family == AF_INET6) {
+		proto = ETHER_TYPE_IPv6;
+		gw_fib_id = lpm_lookup_ipv6(ltbl->lpm6, update->gw.v6.s6_addr);
+		if (gw_fib_id < 0)
+			return -1;
+		gw_fib = &ltbl->fib_tbl6[gw_fib_id];
+
+		if (inet_ntop(AF_INET6, &update->ip.v6.s6_addr,
+				ip_buf, sizeof(ip_buf)) == NULL)
+			return -1;
+
+		ret = snprintf(ipp_buf, sizeof(ipp_buf), "%s/%hhu",
+			ip_buf, update->prefix_len);
+		RTE_VERIFY(ret > 0 && ret < (int)sizeof(ipp_buf));
+	} else {
+		proto = ETHER_TYPE_IPv4;
```

Comment:
Let's put IPv4 first, and use the structure `else if (likely(update->family == AF_INET6)) ... else { ...`